jupyter-org-extensions.el

```diff
@@ -0,0 +1,411 @@
+    (beginning-of-line)
+    (let ((kernelspec (jupyter-completing-read-kernelspec)))
+      (insert
+       (org-element-interpret-data
+        (org-element-put-property
+         (jupyter-org-src-block
+             (format "jupyter-%s" (plist-get (cddr kernelspec) :language)) "" "\n")
+         :post-blank 0))))
+    (forward-line -2)))
+
+;;;###autoload
+(defun jupyter-org-split-src-block (&optional below)
+  "Split the current src block with point in upper block.
+
+With a prefix BELOW move point to lower block."
+  (interactive "P")
+  (unless (org-in-src-block-p)
+    (error "Not in a source block"))
+  (beginning-of-line)
+  (org-babel-demarcate-block)
+  (if below
+      (progn
+        (org-babel-next-src-block)
+        (forward-line)
+        (open-line 1))
+    (forward-line -2)
+    (end-of-line)))
+
+;;;###autoload
+(defun jupyter-org-execute-and-next-block (&optional new)
+  "Execute his block and jump or add a new one.
+
+If a new block is created, use the same language, switches and parameters.
+With prefix arg NEW, always insert new cell."
+  (interactive "P")
+  (unless (org-in-src-block-p)
+    (error "Not in a source block"))
+  (let ((next-src-block
+         (save-excursion (ignore-errors (org-babel-next-src-block)))))
+    ;; instert a new block before executing the current block; otherwise, the new
+    ;; ... block gets added to the results of the next block (due to how
+    ;; ... jupyter works)
+    (when (or new (not next-src-block))
+      (save-excursion
+        (jupyter-org-insert-src-block t)))
+    (org-babel-execute-src-block)
+    (org-babel-next-src-block)))
+
+;;;###autoload
+(defun jupyter-org-execute-to-point ()
+  "Execute all the src blocks that start before point."
+  (interactive)
+  (let ((p (point)))
+    (save-excursion
+      (goto-char (point-min))
+      (while (and (org-babel-next-src-block) (< (point) p))
+        (org-babel-execute-src-block)))))
+
+;;;###autoload
+(defun jupyter-org-restart-kernel-execute-block ()
+  "Restart the kernel of the source block where point is and execute it."
+  (interactive)
+  (jupyter-org-with-src-block-client
+   (jupyter-repl-restart-kernel))
+  (org-babel-execute-src-block-maybe))
+
+;;;###autoload
+(defun jupyter-org-restart-and-execute-to-point ()
+  "Kill the kernel and run src-blocks to point."
+  (interactive)
+  (jupyter-org-with-src-block-client
+   (jupyter-repl-restart-kernel))
+  (jupyter-org-execute-to-point))
+
+;;;###autoload
+(defun jupyter-org-restart-kernel-execute-buffer ()
+  "Restart kernel and execute buffer."
+  (interactive)
+  (jupyter-org-with-src-block-client
+   (jupyter-repl-restart-kernel))
+  (org-babel-execute-buffer))
+
+;;;###autoload
+(defun jupyter-org-jump-to-block (&optional N)
+  "Jump to a block in the buffer.
+If narrowing is in effect, only a block in the narrowed region.
+Use a numeric prefix N to specify how many lines of context to use.
+Defaults to 3."
```

Comment:
I think it would be better to have this as a customizable variable. Something like
`jupyter-org-jump-to-block-context-lines` with doc string 
``Default number of lines to use for context in `jupyter-org-jump-to-block'``.

Should be good to go for merging after this!

Reply:
Not a bad idea.